x-pack/plugins/cases/server/services/user_actions/transform.ts

```diff
@@ -0,0 +1,343 @@
+  }
+
+  return {
+    transformedActionDetails: JSON.stringify(transformedActionDetails),
+    references: referencesToReturn,
+  };
+}
+
+function isCreateCaseConnector(
+  action: string,
+  actionFields: string[],
+  actionDetails: unknown
+): actionDetails is { connector: CaseConnector } {
+  try {
+    const unsafeCase = actionDetails as CaseAttributes;
+
+    return (
+      isCreateConnector(action, actionFields) &&
+      unsafeCase.connector !== undefined &&
+      CaseConnectorRt.is(unsafeCase.connector)
+    );
+  } catch {
+    return false;
+  }
+}
+
+export const ConnectorIdReferenceName: Record<UserActionFieldType, ConnectorIdRefNameType> = {
+  [UserActionFieldType.New]: CONNECTOR_ID_REFERENCE_NAME,
+  [UserActionFieldType.Old]: USER_ACTION_OLD_ID_REF_NAME,
+};
+
+interface ExtractedConnector {
+  transformedActionDetails: unknown;
+  references: SavedObjectReference[];
+}
+
+function transformCreateConnector(
+  connector: CaseConnector,
+  fieldType: UserActionFieldType
+): {
+  transformedActionDetails: { connector: unknown };
+  references: SavedObjectReference[];
+} {
+  const { transformedConnector, references } = transformConnectorIdToReference(
+    ConnectorIdReferenceName[fieldType],
+    connector
+  );
+
+  return {
+    transformedActionDetails: transformedConnector,
+    references,
+  };
+}
+
+type ConnectorIdRefNameType =
+  | typeof CONNECTOR_ID_REFERENCE_NAME
+  | typeof USER_ACTION_OLD_ID_REF_NAME;
+
+export const transformConnectorIdToReference = (
+  referenceName: ConnectorIdRefNameType,
+  connector?: {
+    id?: string;
+  }
+): {
+  transformedConnector: { connector: unknown };
+  references: SavedObjectReference[];
+} => {
+  const { id: connectorId, ...restConnector } = connector ?? {};
+
+  const references = createConnectorReference(connectorId, ACTION_SAVED_OBJECT_TYPE, referenceName);
+
+  const { id: ignoreNoneId, ...restNoneConnector } = getNoneCaseConnector();
+  const connectorFieldsToReturn =
+    connector && isConnectorIdValid(connectorId) ? restConnector : restNoneConnector;
+
+  return {
+    transformedConnector: {
+      connector: connectorFieldsToReturn,
+    },
+    references,
+  };
+};
+
+const createConnectorReference = (
+  id: string | null | undefined,
+  type: string,
+  name: string
+): SavedObjectReference[] => {
+  return isConnectorIdValid(id)
+    ? [
+        {
+          id,
+          type,
+          name,
+        },
+      ]
+    : [];
+};
+
+const isConnectorIdValid = (id: string | null | undefined): id is string =>
+  id != null && id !== noneConnectorId;
+
+function isUpdateCaseConnector(
+  action: string,
+  actionFields: string[],
+  actionDetails: unknown
+): actionDetails is CaseConnector {
+  try {
+    return isUpdateConnector(action, actionFields) && CaseConnectorRt.is(actionDetails);
+  } catch {
+    return false;
+  }
+}
+
+function transformUpdateConnector(
```

Comment:
Talked with Christos and we think we could remove this and use `transformCreateConnector` instead.